zip-0244.rst

```diff
@@ -490,29 +490,47 @@ Identical to that specified for the transaction identifier.
 
 S.2: transparent_sig_digest
 ```````````````````````````
-If we are producing a hash for the signature over a Sapling Spend or an Orchard Action,
-the value of ``transparent_sig_digest`` is identical to the value specified in section
-`T.2 <#t-2-transparent-digest>`_.
-
-If we are producing a hash for signature over a transparent input, the value of
-``transparent_sig_digest`` depends upon the value of a ``hash_type`` flag as in ZIP 143
-[#zip-0143]_.
+In the case that transparent inputs or outputs are present, the value of
+``transparent_sig_digest`` depends upon the value of a ``hash_type`` flag.
 
 The construction of each component below depends upon the values of the
-``hash_type`` flag bits. Each component will be described separately
+``hash_type`` flag bits. Each component will be described separately.
 
 This digest is a BLAKE2b-256 hash of the following values ::
 
-   S.2a: prevouts_sig_digest (32-byte hash)
-   S.2b: sequence_sig_digest (32-byte hash)
-   S.2c: outputs_sig_digest  (32-byte hash)
-   S.2d: txin_sig_digest     (32-byte hash)
+   S.2a: hash_type               (1 byte)
+   S.2b: prevouts_sig_digest     (32-byte hash)
+   S.2c: amounts_sig_digest      (32-byte hash)
+   S.2d: script_codes_sig_digest (32-byte hash)
+   S.2e: sequence_sig_digest     (32-byte hash)
+   S.2f: outputs_sig_digest      (32-byte hash)
+   S.2g: txin_sig_digest         (32-byte hash)
 
 The personalization field of this hash is set to::
 
    "ZTxIdTranspaHash"
 
-S.2a: prevouts_sig_digest
+In the case that the transaction has no transparent components,
+``transparent_sig_digest`` is ::
+
+  BLAKE2b-256("ZTxIdTranspaHash", [])
+
+S.2a: hash_type
+'''''''''''''''
+This is an 8-bit unsigned value. The ``SIGHASH`` encodings from the legacy
+script system are reused: one of ``SIGHASH_ALL``, ``SIGHASH_NONE``, and
+``SIGHASH_SINGLE``, with or without the ``SIGHASH_ANYONECANPAY`` flag. The
+following restrictions apply, which cause validation failure if violated:
+
+- Using any undefined ``hash_type`` (not 0x01, 0x02, 0x03, 0x81, 0x82, or 0x83).
+- Using ``SIGHASH_SINGLE`` without a "corresponding output" (an output with the
+  same index as the input being verified).
+
+If we are producing a hash for the signature over a Sapling Spend or an Orchard
+Action, ``hash_type`` is set to ``SIGHASH_ALL``, and is not encoded anywhere in
```

Comment:
I find this confusing. What does "is set to" pertain to?

Reply:
I don't know how to write this in any plainer English: `hash_type` is set to `SIGHASH_ALL`. That is, this 1-byte input to the transparent sig digest is set to a constant, rather than taken from the transaction encoding.

Reply:
Maybe an addendum: "hash_type is set to SIGHASH_ALL and the the hashes `S.2b..S.2g` are computed in accordance with this flag value."

I think it's the "is not encoded anywhere" that threw me off - it's not encoded, but it is committed to in `S.2a`?

Reply:
Resolved by suggested changes commit